ibm/data_source_ibm_database.go

```diff
@@ -0,0 +1,429 @@
+package ibm
+
+import (
+	"fmt"
+	"github.com/IBM-Cloud/bluemix-go/bmxerror"
+	"github.com/IBM-Cloud/bluemix-go/models"
+
+	"github.com/IBM-Cloud/bluemix-go/api/icd/icdv4"
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/controller"
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/management"
+
+	"github.com/hashicorp/terraform/flatmap"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceIBMDatabaseInstance() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIBMDatabaseInstanceRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Resource instance name for example, my Database instance",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+
+			"resource_group_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The id of the resource group in which the Database instance is present",
+			},
+
+			"location": {
+				Description: "The location or the environment in which Database instance exists",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+
+			"service": {
+				Description: "The name of the Cloud Internet database service",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"plan": {
+				Description: "The plan type of the Database instance",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+
+			"status": {
+				Description: "The resource instance status",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"adminuser": {
+				Description: "The admin user id for the instance",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"adminpassword": {
+				Description: "The admin user id for the instance",
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+			},
+			"version": {
+				Description: "The database version to provision if specified",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"members_memory_allocation_mb": {
+				Description: "Memory allocation required for cluster",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			"members_disk_allocation_mb": {
+				Description: "Disk allocation required for cluster",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			"users": {
```

Comment:
Is this field optional or computed?

Reply:
Users and memory etc are all computed for this data_source. However it lead me to thinking about what was optional and what was not. The schema is OK, but there is a dependency on the provider region parameter being set correctly. The default is us-south and if the instance is not in us-south both the data_source create and import will fail. I've updated the doc to highlight this for both the data_source and resource. Also the error messages if the db instance is not found. The instance is found via the Resource Controller in any region, but the resource specific queries using the ICD API will fail if the provider is not set correctly.